backend/src/main/java/com/bakdata/conquery/mode/local/UpdateMatchingStatsSqlJob.java

```diff
@@ -0,0 +1,352 @@
+	) {
+		final Table<Record> connectorTable = table(name(connector.getResolvedTable().getName()));
+		final Set<Field<?>> connectorColumns = relevantColumns.get(connector);
+		final Field<Object> primaryKey = TablePrimaryColumnUtil.findPrimaryColumn(connector.getResolvedTable(), databaseConfig).as(ENTITIES);
+
+		// we have to select all possible validity dates of all connectors because we have to union multiple connectors
+		final List<Field<?>> validityDates =
+				validityDateMap.entrySet().stream()
+							   .flatMap(entry -> entry.getValue().stream().map(columnDateRange -> entry.getKey() == connector
+																								  ? columnDateRange
+																								  : functionProvider.nulled(columnDateRange))
+													  .flatMap(columnDateRange -> columnDateRange.toFields().stream()))
+							   .toList();
+
+		// connector might have a condition
+		final Condition connectorCondition = toJooqCondition(connector, Optional.ofNullable(connector.getCondition()));
+
+		return dslContext.select(primaryKey)
+						 .select(connectorColumns)
+						 .select(validityDates)
+						 .from(connectorTable)
+						 .where(connectorCondition);
+	}
+
+	private Condition toJooqCondition(final Connector connector, final Optional<CTCondition> childCondition) {
+		final CTConditionContext context = CTConditionContext.create(connector, functionProvider);
+		return childCondition.or(() -> Optional.ofNullable(connector.getCondition()))
+							 .map(condition -> condition.convertToSqlCondition(context).condition())
+							 .orElse(noCondition());
+	}
+
+	/**
+	 * Select the minimum of the least start date and the maximum of the greatest end date of all validity dates of all connectors.
+	 */
+	private Field<String> toValidityDateExpression(final Map<Connector, List<ColumnDateRange>> validityDateMap) {
+		final List<ColumnDateRange> validityDates = validityDateMap.values().stream().flatMap(List::stream).map(functionProvider::toDualColumn).toList();
+		final List<Field<Date>> allStarts = validityDates.stream().map(ColumnDateRange::getStart).toList();
+		final List<Field<Date>> allEnds = validityDates.stream().map(ColumnDateRange::getEnd).toList();
+		final ColumnDateRange minAndMax = ColumnDateRange.of(min(functionProvider.least(allStarts)), max(functionProvider.greatest((allEnds))));
+		return functionProvider.daterangeStringExpression(minAndMax);
+	}
+
+	private void mapRecordToConceptElements(
+			final TreeConcept treeConcept,
+			final Record record,
+			final List<Field<?>> relevantColumns,
+			final ConceptTreeCache treeCache
+	) {
+		final CalculatedValue<Map<String, Object>> rowMap = new CalculatedValue<>(record::intoMap);
+		final MatchingStats.Entry entry = toMatchingStatsEntry(record);
+
+		if (treeConcept.getChildren().isEmpty()) {
+			addEntryToConceptElement(treeConcept, treeConcept.getName(), entry);
+			return;
+		}
+
+		relevantColumns.stream().map(field -> record.get(field, String.class)).forEach(relevantColumnValue -> {
```

Comment:
Du musst nicht über die relevantColumns iterieren. Es wird nur die conceptColumn in findMostSpecificChild gegeben. Die anderen werden aber über die rowMap abgegriffen, deswegen klappt das mit dem group-by

Reply:
Ok, ich verstehe warum, aber dann muss ich die Logik nochmal umbauen. Es kann ja sein, dass es keine Connector Column gibt. Das schaue ich mir nochmal an.

Reply:
Ah lol, ich habe einfach den alten Stand lokal angeguckt 😄